src/chart/graph/circularLayoutHelper.ts

```diff
@@ -77,6 +79,25 @@ export function circularLayout(
         return;
     }
 
+    if (draggingNode) {
+        const [tempX, tempY] = draggingNode.getLayout();
+        const tan = (tempY - cy) / (tempX - cx);
+        const radian = -Math.atan(tan);
+
+        tempX > cx
```

Comment:
There is a better way to fix the node on the circle edge.

```ts
import * as vector from 'zrender/src/core/vector';
// Vector from circle center to node.
const v = [tempX - cx, tempY - cy];
// Normalize the vector from circle center to node.
vector.normalize(v, v);
// Scale to circle radius
vector.scale(v, v, r);
// Calcualate the new position
draggingNode.setLayout([cx + v[0], cy + v[1]]); 
```

Reply:
done~